mergify_engine/tests/functional/test_subscription.py

```diff
@@ -0,0 +1,47 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2018—2021 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from mergify_engine import config
+from mergify_engine import context
+from mergify_engine import subscription
+from mergify_engine.tests.functional import base
+
+
+class TestSubscription(base.FunctionalTestBase):
+    async def test_subscription(self):
+        self.subscription = subscription.Subscription(
+            self.redis_cache,
+            config.TESTING_ORGANIZATION_ID,
+            self.SUBSCRIPTION_ACTIVE,
+            "Abuse !",
+            frozenset(
+                getattr(subscription.Features, f)
+                for f in subscription.Features.__members__
+            )
+            if self.SUBSCRIPTION_ACTIVE
+            else frozenset([subscription.Features.PUBLIC_REPOSITORY]),
+        )
+
+        await self.subscription._save_subscription_to_cache()
+        await self.setup_repo()
+        pr, _ = await self.create_pr()
+        await context.Context.create(self.repository_ctxt, pr)
+        await self.run_engine()
+
+        assert (await self.get_pull(pr["number"]))["state"] == "open"
+
+        expected = "Abuse !"
+        assert expected == self.subscription.reason
```

Comment:
This only check the reason on the subscription object.
What we want is to check this reason has been posted in a  check_runs on GitHub.

Reply:
You can take a look at this test: https://github.com/Mergifyio/mergify-engine/blob/main/mergify_engine/tests/functional/test_engine.py#L41-L54